src/python/pants/backend/python/subsystems/subprocess_environment.py

```diff
@@ -0,0 +1,57 @@
+# coding=utf-8
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import os
+
+from pants.engine.rules import optionable_rule, rule
+from pants.subsystem.subsystem import Subsystem
+from pants.util.objects import datatype, string_optional
+
+
+class SubprocessEnvironment(Subsystem):
+  options_scope = 'subprocess-environment'
+
+  @classmethod
+  def register_options(cls, register):
+    super(SubprocessEnvironment, cls).register_options(register)
+
+    # TODO(#7735): move the --lang and --lc-all flags to a general subprocess support subystem.
+    register('--lang',
+             default=os.environ.get('LANG'),
+             fingerprint=True, advanced=True,
+             help='Override the `LANG` environment variable for any forked subprocesses.')
+    register('--lc-all',
+             default=os.environ.get('LC_ALL'),
+             fingerprint=True, advanced=True,
+             help='Override the `LC_ALL` environment variable for any forked subprocesses.')
+
+
+class SubprocessEncodingEnvironment(datatype([
```

Comment:
I'm not sure that the datatype is necessary here (could just consume the Subsystem?) but not a blocker.